src/test/integration/tests/Deposit_Delegate_Allocate.t.sol

```diff
@@ -189,11 +275,178 @@ contract Integration_Deposit_Delegate_Allocate is IntegrationCheckUtils {
+
+        // 7. Complete withdrawal
+        _rollBlocksForCompleteWithdrawals(withdrawals);
+        for (uint256 i = 0; i < withdrawals.length; ++i) {
+            uint256[] memory expectedTokens =
+                _calculateExpectedTokens(withdrawals[i].strategies, withdrawals[i].scaledShares);
+            staker.completeWithdrawalAsTokens(withdrawals[i]);
+            check_Withdrawal_AsTokens_State_AfterSlash(
+                staker, operator, withdrawals[i], allocateParams, slashingParams, expectedTokens
+            );
+        }
+        
+        // Check Final State
+        assert_HasUnderlyingTokenBalances(
+            staker,
+            allocateParams.strategies,
+            tokenBalances,
+            "staker should have withdrawn all shares"
+        );
+        assert_NoWithdrawalsPending(withdrawalRoots, "all withdrawals should be removed from pending");
+    }
 
     function testFuzz_deposit_delegate_allocate_deregister_slash(
         uint24 _random
-    ) public {}
+    ) public {
+        _configRand({_randomSeed: _random, _assetTypes: HOLDS_ALL, _userTypes: DEFAULT});
+        _upgradeEigenLayerContracts(); // Upgrade contracts if forkType is not local
+
+        (User staker, IStrategy[] memory strategies, uint256[] memory tokenBalances) = _newRandomStaker();
+        (User operator,,) = _newRandomOperator();
+        (AVS avs,) = _newRandomAVS();
+
+        // 1. Deposit Into Strategies
+        staker.depositIntoEigenlayer(strategies, tokenBalances);
+        // 2. Delegate to an operator
+        staker.delegateTo(operator);
+
+        // Create an operator set and register an operator.
+        OperatorSet memory operatorSet = avs.createOperatorSet(strategies);
+        operator.registerForOperatorSet(operatorSet);
+
+        // 3. Allocate to operator set.
+        IAllocationManagerTypes.AllocateParams memory allocateParams =
+            operator.modifyAllocations(operatorSet, _randMagnitudes({sum: 1 ether, len: strategies.length}));
+        assert_Snap_Allocations_Modified(
+            operator, allocateParams, false, "operator allocations should be updated before delay"
+        );
+        _rollBlocksForCompleteAllocation(operator, operatorSet, strategies);
+        assert_Snap_Allocations_Modified(
+            operator, allocateParams, true, "operator allocations should be updated after delay"
+        );
+
+        // 4. Deregister.
+        operator.deregisterFromOperatorSet(operatorSet);
+
+        // 5. Slash operator
+        IAllocationManagerTypes.SlashingParams memory slashingParams;
+        {
+            (IStrategy[] memory strategiesToSlash, uint256[] memory wadsToSlash) =
+                _randStrategiesAndWadsToSlash(operatorSet);
+            slashingParams = avs.slashOperator(operator, operatorSet.id, strategiesToSlash, wadsToSlash);
+            assert_Snap_Allocations_Slashed(slashingParams, operatorSet, true, "operator allocations should be slashed");
```

Comment:
Don't we also want to assert shares?

Reply:
fixed in [a32af54](https://github.com/Layr-Labs/eigenlayer-contracts/pull/961/commits/a32af541609aea5e4685a4bdd765dfde5925f89f)